jdaviz/configs/default/plugins/data_menu/data_menu.py

```diff
@@ -245,6 +248,63 @@ def _on_app_icons_updated(self, msg):
             self.layer_icons = msg.icons
         self._set_viewer_id()
 
+    def _on_layers_finalized(self, msg=None):
+        # when a catalog is added, set visibility based on alignment type
+        # and presence of pixel and/or world coordinates.
+        new_data = msg.data
+        if new_data.meta.get('_importer') == 'CatalogImporter':
+            self._handle_catalog_layer_visibility(layers=[new_data])
+
+    def _on_link_changed(self, msg=None):
+
+        if len(self._viewer.layers):
+            layers = [layer.layer for layer in self._viewer.layers]
```

Comment:
I think you need the actual data, not the layer, here.  Either `layer.data` or `layer.layer.data` (can't ever remember without checking)

Reply:
i verified that layer is needed here, not layer.data